drivers/ethernet/eth_sam_gmac.c

```diff
@@ -1918,15 +1961,16 @@ static void eth0_iface_init(struct net_if *iface)
 		LOG_ERR("ETH PHY Initialization Error");
 		return;
 	}
-	/* PHY auto-negotiate link parameters */
-	result = phy_sam_gmac_auto_negotiate(&cfg->phy, &link_status);
-	if (result < 0) {
-		LOG_ERR("ETH PHY auto-negotiate sequence failed");
-		return;
-	}
 
-	/* Set up link parameters */
-	link_configure(cfg->regs, link_status);
+	/* Initialise monitor */
+	k_work_init(&dev_data->monitor_work, monitor_work_handler);
```

Comment:
Instead of timer followed by work, you could simply use delayed work instead, imho the code would look cleaner that way.

Reply:
The original intention was to submit multiple "monitor task" works from the timer, but on second thought, the monitor will likely not have anything else to do in the future and this does seem unnecessary.

I have removed the timer and re-implemented the process using delayed work.